iml-services/iml-action-runner/src/main.rs

```diff
@@ -49,13 +50,29 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     .map(|x| warp::reply::json(&x))
     .with(log);
 
-    let addr = unsafe { NetUnixListener::from_raw_fd(3) };
-
-    let listener = UnixListener::try_from(addr)?
-        .incoming()
-        .inspect_ok(|_| tracing::debug!("Client connected"));
-
-    tokio::spawn(warp::serve(routes).serve_incoming(listener));
+    match env::var("LISTEN_PID") {
```

Comment:
Instead of having an `LISTEN_PID` env var that we must supply, we should just check the raw file handle, and if that fails, try the `ACTION_RUNNER_PORT`.

Reply:
Sure. I did it this way to be consistent with the iml-update-check repo. My understanding (after testing locally) is that the LISTEN_PID environment variable is set when the socket is established (I tested this with netcat). If the socket isn't activated then LISTEN_PID isn't set. In either case, i'll update to check the raw file handle.

Reply:
No need, I'm abstracting that bit in a separate patch. one sec

Reply:
sounds good.

Reply:
https://github.com/whamcloud/integrated-manager-for-lustre/pull/1336/